WIP: add towncrier integration (#1979)
ansible/molecule: changelog/README.rst

*********************
Manging the Changelog
*********************

This directory contains "newsfragments" which are short files that contain a

> small **ReST**-formatted text that will be added to the next ``CHANGELOG``.

Reviewer: Can we please use `ReStructuredText` term? https://en.wikipedia.org/wiki/ReStructuredText
Google search for `ReST` gives a lot of misleading results. Even better: maybe link it to do SphinX docs page.

The ``CHANGELOG`` will be read by users, so this description should be aimed to

> molecule users instead of describing internal changes which are only relevant

Reviewer: I think internal changes can still be useful. I saw others putting those under a separate category, like `misc` or `internal`.

OTOH wrt the style, we should encourage fragment authors to follow the same principles as are useful for docstrings in Python or commit messages in Git. For example, use imperative sentences. Describe things on the high-level outlining effects of the changes, not implementation details.

Author: Fair points. 

However, for the first attempt, let's just deal with our main priority: a user changelog.

to the developers.

Make sure to use full sentences with correct case and punctuation, for

> example::

Reviewer: Maybe have explicit markup highlighting?
```suggestion
example:

.. code:: rst
```

>     Fix issue with non-ascii messages from the ``warnings`` module.

Reviewer: This example doesn't give the reader an understanding of what the issue was. It's kinda blurry. I think it might be better to re-phrase it as:
```suggestion
    Fix broken output of non-ascii messages produced by the ``warnings`` module.
```

In some projects (tox?) they also encourage tagging authors of the change in their template.

Reviewer: I'd also like @acozine about the best wording as it's docs-related.

Each file should be named like ``<ISSUE>.<TYPE>.rst``, where ``<ISSUE>`` is an
issue number, and ``<TYPE>`` is one of:

* ``feature``: new user facing features, like new command-line options and new behavior.
* ``bugfix``: fixes a reported bug.
* ``doc``: documentation improvement, like rewording an entire session or adding missing docs.
* ``deprecation``: feature deprecation.
* ``removal``: feature removal.
* ``trivial``: fixing a small typo or internal change that might be noteworthy.

So for example: ``123.feature.rst``, ``456.bugfix.rst``.

If your PR fixes an issue, use that number here. If there is no issue, then
after you submit the PR and get the PR number you can add a changelog using
that instead.

If you are not sure what issue type to use, don't hesitate to ask in your PR.

``towncrier`` preserves multiple paragraphs and formatting (code blocks, lists,
and so on), but for entries other than ``features`` it is usually better to
stick to a single paragraph to keep it concise. You can install ``towncrier``

> and then run ``towncrier --draft`` if you want to get a preview of how your

Reviewer: So for this, we'll probably need to have some tox envs instead of asking user to do manual steps
